### PR TITLE
Refactor security policies

### PR DESCRIPTION
**Pull request by janmazak**

### CODE REVIEW

File: src/keyDerivation.c [hunk 1]
```diff
@@ -11,42 +11,20 @@
...
 )
 {
-	if (!_isPathAllowed(pathSpec)) {
-		THROW(ERR_INVALID_BIP44_PATH);
-	}
 	// Sanity check
 	ASSERT(pathSpec->length <= ARRAY_LEN(pathSpec->path));
 
+	// if the path is invalid, it's a bug in previous validation
+	ASSERT(policyForDerivePrivateKey(pathSpec) != POLICY_DENY);
```
Comment (mkv-vcm) [inline src/keyDerivation.c:26]: ensure_not_denied? Just for consistency? Or we don't want to throw, we want to assert here?

**Review by mkv-vcm — COMMENTED** [review]
How do we decide what is a security policy and what is a data validation?

File: src/keyDerivation.c (hunk 1 above)
Comment (janmazak) [reply src/keyDerivation.c:26]: Yes, just an assert, the path should have been validated previously.

**Comment by janmazak** [comment]
Generally, the difference between validation and security policies is that data deemed valid could be serialized into a valid tx, but it is not safe to do so from the perspective of the HW wallet user (e.g. he is not aware what he is signing). 

This is not that straightforward to implement in Ledger, so in some places we might deviate from this basic principle.

**Review by mkv-vcm — APPROVED** [review]

**Pull request closed without merging**